rskj-core/src/main/java/co/rsk/core/Wallet.java

```diff
@@ -193,7 +195,9 @@ public RskAddress addAccountWithPrivateKey(byte[] privateKeyBytes, String passph
 
     private void saveAccount(Account account) {
         synchronized (accessLock) {
-            accounts.put(account.getAddress(), account.getEcKey().getPrivKeyBytes());
+            if (!accounts.containsKey(account.getAddress())) {
```

Comment:
this seems unnecessary, why don't just put the value on the same address? wouldn't it always be the same deriving same priv?

Reply:
@ajlopez I just tried removing this `if` and seems fine

Reply:
It is possible to avoid this validation and just have the other. I don not know so far the code so I prefer to wait until @ajlopez comments about this.